paywall/src/unlock.js/PostMessageEmitters/CheckoutIframeMessageEmitter.ts

```diff
@@ -42,29 +43,49 @@ export default class CheckoutIframeMessageEmitter extends FancyEmitter {
 
   public readonly postMessage: PostMessageResponder<PostMessages>
   public readonly iframe: IframeType
+  private buffer: Array<any>
+  private isReady: boolean
 
   constructor(
     window: IframeManagingWindow & PostOfficeWindow & OriginWindow,
     checkoutIframeUrl: string
   ) {
     super()
-
+    this.isReady = false
     this.window = window
     this.iframe = makeIframe(window, checkoutIframeUrl, 'unlock checkout')
     const url = new URL(this.iframe.src)
     addIframeToDocument(window, this.iframe)
 
+    this.buffer = []
+
     const { postMessage, addHandler } = mainWindowPostOffice(
       window,
       this.iframe,
       url.origin,
       'main window',
       'Checkout UI iframe'
     )
-    this.postMessage = postMessage
+    // We want to only post message when we're ready!
+    this.postMessage = (type, payload) => {
```

Comment:
For now we are sending messages even if the iframe is not loaded. this logic adds sime kind of buffering to make sure that the iframe is ready before sending messages.